src/Helper/GridHelperService.php

```diff
@@ -591,6 +591,18 @@ public function prepareListingForGrid(array $requestParams, string $requestedLan
             $conditionFilters[] = '(`path` = ' . $quotedPath . ' OR `path` like ' . $quotedWildcardPath . ')';
         }
 
+        if (isset($requestParams['all_objects']) && $requestParams['all_objects'] === 'true') {
+            $conditionFilters[] = "`type` = 'object' OR `type` = 'variant'";
+        } else {
+            if (isset($requestParams['only_variant_objects']) && $requestParams['only_variant_objects'] === 'true') {
+                $conditionFilters[] = "`type` = 'variant'";
+            } elseif(isset($requestParams['only_objects']) && $requestParams['only_objects'] === 'true') {
+                $conditionFilters[] = "`type` = 'object'";
+            } else {
+                $conditionFilters[] = "`type` = 'object' OR `type` = 'variant'";
```

Comment:
```suggestion
            $conditionFilters[] = "(`type` = 'object' OR `type` = 'variant')";
        } else {
            if (isset($requestParams['only_variant_objects']) && $requestParams['only_variant_objects'] === 'true') {
                $conditionFilters[] = "`type` = 'variant'";
            } elseif(isset($requestParams['only_objects']) && $requestParams['only_objects'] === 'true') {
                $conditionFilters[] = "`type` = 'object'";
            } else {
                $conditionFilters[] = "(`type` = 'object' OR `type` = 'variant')";
```
See also https://github.com/pimcore/admin-ui-classic-bundle/blob/fa2d03aaeb4787e1813f71f6a7ec58b28cd820cb/src/Helper/GridHelperService.php#L638

because otherwise everything is truthy when `type` is `variant`.